samples/sunlight_model/src/isc.c

```diff
@@ -0,0 +1,334 @@
+    }
+    pkt.message_info.size = written_size;
+    zt_isc_net_pkt_calc_crc(&pkt);
+    zt_isc_net_pkt_send(&pkt);
+    return test_failed;
+}
+
+void ISC_TX_service_callback(zt_channel_e id)
+{
+    k_msgq_put(&channel_queue, &id, K_NO_WAIT);
+}
+
+void ISC_service_callback(zt_channel_e id)
+{
+}
+
+void ISC_TX_task(void)
+{
+    zt_channel_e id = ZT_CHANNEL_COUNT;
+    while (1) {
+        if (!k_msgq_get(&channel_queue, &id, K_FOREVER)) {
+            switch (id) {
+            case ZT_LIGHT_STATUS_CHANNEL: {
+                struct zt_isc_net_pkt pkt =
+                    ZT_ISC_NET_PKT_WITH_MSG_SIZE(sizeof(zt_data_light_status_msg_t));
+                pkt.header.op           = OP_UPDATE;
+                pkt.header.channel      = id;
+                zt_data_t *light_status = ZT_DATA_LIGHT_STATUS_MSG(0);
+                zt_chan_read(ZT_LIGHT_STATUS_CHANNEL, light_status);
+                int err = zt_isc_net_pkt_set_message(&pkt, (char *) light_status,
+                                                     sizeof(zt_data_light_status_msg_t));
+                if (err) {
+                    printk("Error set %d\n", err);
+                }
+                zt_isc_net_pkt_send(&pkt);
+                break;
+            }
+            default: {
+                printk("D: tx task invalid channel");
+            }
+            }
+        }
+    }
+}
+
+void ISC_task(void)
+{
+    char dev_name[] = UART_DEVICE_NAME;
+    if (uart_open(dev_name)) {
+        return;
+    }
+
+    if (SELF_TEST()) {
+        return;
+    }
+
+    char data                 = 0;
+    struct zt_isc_net_pkt pkt = ZT_ISC_NET_PKT_WITH_MSG_SIZE(255);
+    pkt.header.op             = OP_DEBUG;
+    zt_isc_net_pkt_set_message(&pkt, "hello world", sizeof("hello world!"));
+
+    while (1) {
+        if (!k_msgq_get(uart_get_input_msgq(), &data, K_FOREVER)) {
+            if (digest_byte(data, &pkt)) {
+                /* Process the pkt */
+                switch (pkt.header.op) {
+                case OP_WRITE:
+                case OP_UPDATE: {
+                    if (pkt.header.channel == ZT_SUNLIGHT_LEVEL_CHANNEL) {
+                        zt_data_t *s = (zt_data_t *) pkt.message;
+                        int err      = zt_chan_pub(pkt.header.channel, s);
+                        if (err) {
+                            printk("publishing status %d\n", err);
+                        }
+                        // printk("Size = %u, level = %d\n", s->sunlight_level_msg.size,
+                        //        s->sunlight_level_msg.value.level);
+                        // printk("write or update\n");
+                        // for (int i = 0; i < pkt.message_info.size; ++i) {
+                        //     printk("%02X ", pkt.message[i]);
+                        // }
+                        // printk("\n");
+                    }
+                }
+                }
+                // zt_isc_net_pkt_send(&pkt);
```

Comment:
The same here.

Reply:
The whole file is not being used in fact. It was the initial ISC's implementation. Removed from the base.